app/messages/list_message.rb

```diff
@@ -25,6 +27,7 @@ def initialize(params={})
       params = params.symbolize_keys
       @pagination_params = params.slice(*ALLOWED_PAGINATION_KEYS)
       @pagination_options = PaginationOptions.from_params(params)
+      invalid_param!(@pagination_options.errors.full_messages) unless @pagination_options.valid?
```

Comment:
A Message class shouldn't need to require a controller or have access to the `invalid_param!` method.  Generally, messages should be informative while the controller actually decides what to do with any errors reported by the message.  Please see https://github.com/cloudfoundry/cloud_controller_ng/wiki/Notes-on-V3-Architecture#controllers for more details

Reply:
That's really useful to know, thanks for the link. Looks like we can just add a validator to the list message class that will call the pagination options validate method and add in any errors found so we don't need to use the `invalid_param` method and it is up to the calling controller to throw the correct exception 👍